pkg/appgw/backendhttpsettings.go

```diff
@@ -169,20 +169,18 @@ func (c *appGwConfigBuilder) getBackendsAndSettingsMap(cbCtx *ConfigBuilderConte
 	for backendID, serviceBackendPairs := range serviceBackendPairsMap {
```

Comment:
At line 127 above, targetPortsResolved := c.resolvePortName(sp.Name, &backendID)
we should use sp.TargetPort instead of sp.Name
sp.TargetPort is supposed to be used to compare with endpoint port name

Reply:
@3quanfeng i think you are correct. Looks like a bug in our code.
@akshaysngupta  can you confirm?

Reply:
that is what you would think.

But turns out that kubernetes matches the serverPortName with the container port after resolving.

Reply:
We actually had to fix this because of this.
https://github.com/Azure/application-gateway-kubernetes-ingress/pull/393

Reply:
If EndpointPort name needs to match the 'name' field in the corresponding ServicePort, can we add such a comment to the code?